Modules/Sources/PointOfSale/Presentation/Settings/POSSettingsHardwareDetailView.swift

```diff
@@ -100,6 +111,36 @@ struct POSSettingsHardwareDetailView: View {
 
 // MARK: - Views
 private extension POSSettingsHardwareDetailView {
+    func paymentsOnboardingView(from onboardingViewContainer: CardPresentPaymentOnboardingViewContainer) -> some View {
+        onboardingViewContainer.configuration.showSupport = {
+            posModel.cancelCardPaymentsOnboarding()
+            showSupport = true
+        }
+
+        return PointOfSaleCardPresentPaymentOnboardingView(
+            viewModel: .init(onboardingViewContainer: onboardingViewContainer,
+                             onDismissTap: {
+                                 posModel.cancelCardPaymentsOnboarding()
+                             }))
+        .onAppear {
+            posModel.trackCardPaymentsOnboardingShown()
+        }
+    }
```

Comment:
Given that this is all duplicative as well, could you take a look at including it in the `cardReaderConnecting` modifier when you look at that?